### DISCUSSION TITLE
WARN: There are cyclic dependencies

### DISCUSSION THREAD

**Discussion by zhendalf** [pnpm/pnpm#3569] [Q&A] [answered]
A more recent version of `pnpm` started showing a warning:
```
 WARN  There are cyclic dependencies
```

Is there a way to get some details on what dependencies are forming a cycle? Is it only warning about dependencies in my workspace, or somewhere in the project dependency tree? Or is it a more general warning about the current state of the universe 😁?

**Comment by zkochan** [comment]
cc @javier-garcia-meteologica

related PR #3537

**Comment by javier-garcia-meteologica** [comment, marked as answer]
Pnpm tries to provide information about what packages are causing the cycle. If `graph-sequencer` (the library used to sort packages) has any information about those packages, it is displayed. In your case, since there is no extra information, it's likely because `graph-sequencer` detected a cycle but it cannot name which packages are causing it.

This kind of dependency cycle is about workspace dependencies, perhaps I should change the warning message to make it clear. Which kind of dependencies? `dependencies`, `optionalDependencies` and `devDependencies`.

**Reply by zhendalf** [reply]
Thank you for the explanation!

**Comment by rxliuli** [comment]
Is there a way to simply print it out, now there is only an error message, but no specific module is printed out, it is difficult for me to locate and fix the error
